File: test/fixtures/analysis/python-svc/src/authz/access.py
"""Access-control module under the `src/` layout (src/authz/access.py).

An integration test imports it as `from authz.access import can_access` —
an absolute import rooted at `src/`, not the project root. The import-graph
resolver must credit this file as tested rather than flag it as a gap.
"""


def can_access(user_id: str, resource: str, roles: list[str]) -> bool:
    if "admin" in roles:
        return True
    return resource in roles

File: test/fixtures/analysis/ts-webapp/src/authz/access.ts
// A plain access-control module — the kind of source file an integration
// test exercises by importing it via the `@/` path alias rather than a deep
// relative path. No HTTP calls (keeps the flow fixture's call count intact).

export interface AccessRequest {
  userId: string;
  resource: string;
}

export function canAccess(req: AccessRequest, roles: string[]): boolean {
  if (roles.includes('admin')) return true;
  return roles.includes(req.resource);
}
